donationTracker/app/src/main/java/com/example/teamsplash/donationtracker/controller/CreateAccountActivity.java

```diff
@@ -47,87 +45,94 @@ public void onClick(View v) {
-        String firstname = firstNameField.getText().toString();
-        String lastname = lastNameField.getText().toString();
-        String email = emailAddressField.getText().toString();;
-        String password = passwordField.getText().toString();
-        String confirmPass = confirmPasswordField.getText().toString();
-
-        // Check if everything during the creation of the account is valid
-
-        boolean cancel = false;
-        View focusView = null;
-
-        // Check for a valid email address.
-        if (TextUtils.isEmpty(email)) {
-            emailAddressField.setError(getString(R.string.error_field_required));
-            focusView = emailAddressField;
-            cancel = true;
-        } else if (!isEmailValid(email)) {
-            emailAddressField.setError(getString(R.string.error_invalid_email));
-            focusView = emailAddressField;
-            cancel = true;
+        // Reset error messages displayed in the registration fields.
+        firstNameField.setError(null);
+        lastNameField.setError(null);
+        emailAddressField.setError(null);
+        passwordField.setError(null);
+        confirmPasswordField.setError(null);
+
+        final String firstName = firstNameField.getText().toString();
+        final String lastName = lastNameField.getText().toString();
+        UserType userType = (UserType) userTypeField.getSelectedItem();
+        final String emailAddress = emailAddressField.getText().toString();
+        final String password = passwordField.getText().toString();
+        final String confirmPassword = confirmPasswordField.getText().toString();
+
+        View firstFieldWithInvalidText = null;
+
+        if (password.isEmpty()) {
+            clearPasswordFields();
+            passwordField.setError(getText(R.string.field_required));
+            firstFieldWithInvalidText = passwordField;
+        } else {
+            if (!(password.equals(confirmPassword))) {
+                clearPasswordFields();
+                passwordField.setError(getText(R.string.the_passwords_were_not_matching));
+                firstFieldWithInvalidText = passwordField;
+            } else {
+                if (!(Model.validatePassword(password))) {
+                    clearPasswordFields();
+                    passwordField.setError(getText(R.string.the_password_was_invalid));
+                    firstFieldWithInvalidText = passwordField;
+                }
+            }
         }
 
-        // Check for a valid password, if the user entered one.
-        if (password.length() < MIN_PASSWORD_LENGTH) {
-            passwordField.setError(getString(R.string.error_invalid_password));
-            focusView = passwordField;
-            cancel = true;
-        }
-        // TODO fix this password verification??? I don't understand???? -Sara
-        /** else if (!isPasswordValid()) {
-            confirmPasswordField.setError(getString(R.string.error_mismatched_password));
```

Comment:
pretty sure string.error_mismatched_password and string.the_passwords_were_not_matching are the same

Reply:
I just cleaned up the strings.